dask/dataframe/core.py

```diff
@@ -3609,6 +3609,13 @@ class DataFrame(_Frame):
     _token_prefix = "dataframe-"
     _accessors = set()
 
+    def __init__(self, dsk, name, meta, divisions):
+        super().__init__(dsk, name, meta, divisions)
+        if hasattr(self, "dask") and name in self.dask.layers:
+            self.dask.layers[name].info["type"] = type(self)
+            self.dask.layers[name].info["divisions"] = divisions
+            self.dask.layers[name].info["chunk_type"] = type(meta)
+
```

Comment:
1. could use the same refactoring as da.Array
2. why nothing for Series?
3. dd does not have anything equivalent to array_plugins

Reply:
1. could use the same refactoring as da.Array - ok, done
2. why nothing for Series? - Good point, I've added `"series_dtypes": {col: meta[col].dtype for col in meta.columns}`
3. dd does not have anything equivalent to array_plugins - This looks like a comment and not a request. Thanks for mentioning it!